### ISSUE TITLE
Should Qz really be mandatory?

### ISSUE THREAD

**Issue by smk78** [canSAS-org/NXcanSAS_examples#2]
On 7/5/2016 at 8:45 Adrian Rennie wrote: 

I agree with this idea that Qz should not be required.  Indeed my 
recollection of the discussions about canSAS treated data were that 
fields that are obligatory should be kept to a minimum although one 
might have recommendations.  The thought of preserving and then 
decompressing on read, large numbers of arrays of zeroes for multiple 
data sets is not really what one should be insisting upon.

Best regards, Adrian

---

On 05/07/2016 at 14:33, Andrew Jackson wrote:

I think implicit Qz=0 is what most of us had assumed - especially given
the examples we worked up in Uppsala.

That doesn’t break the Q definition and is current practice in existing
data representations.

Similarly, if one were to represent specular reflectivity data in this
construct (say as part of a GISANS data set) then you might need only
one component given, with it implicit that the others components were
zero (though that is probably out of scope and muddying the waters).

Andrew

---

On 5 July 2016 at 13:27 Pete Jemian wrote:

I like this answer as it is more practical.  But it is not consistent
with the result of our discussions a few years back.  This page
documents the result of that discussion clearly regarding the two
representations of Q:

http://cansas-org.github.io/canSAS2012/basics.html#definition-of

Can we agree now that (Qx, Qy) is acceptable with the assumption of Qz=0?

Pete

---

On 7/5/2016 at 6:26 AM Tobias Richter wrote:

That's not the idea, unless you know that's correct. Since in my
(admittedly limited) experience Qz is not often calibrated, we should
not require that. I don't think it's in all our examples either.

Regards
Tobias

---

On 5 Jul 2016 at 11:09 a.m. Steve King wrote:

So we just add a Qz array with zeroes?

Steve

---

On 05 July 2016 at 01:10 Pete Jemian wrote:

on the 2D NXcanSAS...h5 data file, need a Qz array to go with Qx and Qy

We have agreed that either |Q| (such as in the 1-D case) or full Q
vector are the two choices.

My quick overview says, otherwise OK

Pete

---

On 7/4/2016 at 5:33 AM Steve King wrote:

I have uploaded a new Mantid NXcanSAS test example (_v3) to Pete’s new
GitHub repo

https://github.com/canSAS-org/NXcanSAS_examples/tree/master/Mantid

The fixes include:
1. Introduced canSAS_class
2. Provided NX_class with equivalent class names. This attribute lives
in parallel to the canSAS_class one.
3. Set dates to ISO-8601 format
4. Change the Q_indices attribute to be an int array 5. SDD which is
part of the detectors is now a floating point instead of a string 6.
change |unit|to |units|

As before, comments welcome!

Steve

**Comment by prjemian** [comment]
On 7/5/2016 9:09 AM, stephen.king@stfc.ac.uk wrote:

> So does that mean the _v3.h5 file I posted yesterday is ok? Thou all art content? :-)

---

On 7/11/2016 5:59 AM, stephen.king@stfc.ac.uk wrote:

> Is it agreed that in fact Qz is optional and therefore the latest version of an NXcanSAS file written by Mantid conforms to the standard?

---

On 7/11/2016 6:15 AM, Jemian, Pete R. wrote:

> That was the consensus.  If the data is trivialand its default is
> obvious, such as Qz=0, omitting it is fine.

---

On 7/11/2016 6:36 AM, stephen.king@stfc.ac.uk wrote:

> Ok. I will have the Mantid ticket closed. And update the SasView one
> (for something to read NXcanSAS!).

---
